### PR TITLE
Add manifest for SolveSpace 3.1 (com.solvespace.Solvespace)

### PR DESCRIPTION
**Pull request by rpavlik**
SolveSpace is a GPL3 3D CAD package. I help with maintaining it.

### CODE REVIEW

File: com.solvespace.SolveSpace.json [hunk 1]
```diff
@@ -0,0 +1,150 @@
+{
+    "$schema": "https://raw.githubusercontent.com/TingPing/flatpak-manifest-schema/master/flatpak-manifest.schema",
```
Comment (hfiguiere) [inline com.solvespace.SolveSpace.json:2]: what is this?

File: com.solvespace.SolveSpace.json [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        },
+        {
+            "name": "cairomm",
+            "config-opts": [
+                "--disable-documentation"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "http://ftp.gnome.org/pub/GNOME/sources/cairomm/1.12/cairomm-1.12.0.tar.xz",
```
Comment (hfiguiere) [inline com.solvespace.SolveSpace.json:76]: please use https url.

File: com.solvespace.SolveSpace.json [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        },
+        {
+            "name": "pangomm",
+            "config-opts": [
+                "--disable-documentation"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "http://ftp.gnome.org/pub/GNOME/sources/pangomm/2.40/pangomm-2.40.2.tar.xz",
```
Comment (hfiguiere) [inline com.solvespace.SolveSpace.json:89]: please use https url.

File: com.solvespace.SolveSpace.json [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        },
+        {
+            "name": "atkmm",
+            "config-opts": [
+                "--disable-documentation"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "http://ftp.gnome.org/pub/GNOME/sources/atkmm/2.28/atkmm-2.28.0.tar.xz",
```
Comment (hfiguiere) [inline com.solvespace.SolveSpace.json:102]: please use https url.

File: com.solvespace.SolveSpace.json [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        {
+            "name": "SolveSpace",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/solvespace/solvespace.git",
+                    "commit": "00533a0fb22c85293d253f998c8d6f30029061ea"
+                }
+            ],
+            "buildsystem": "cmake",
```
Comment (hfiguiere) [inline com.solvespace.SolveSpace.json:141]: cmake-ninja doesn't work?

File: com.solvespace.SolveSpace.json [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+            "name": "libjson-c",
+            "sources": [
+                {
+                    /* 0.15-nodoc doesn't build */
+                    "type": "archive",
+                    "url": "https://s3.amazonaws.com/json-c_releases/releases/json-c-0.13.1-nodoc.tar.gz",
+                    "sha256": "94a26340c0785fcff4f46ff38609cf84ebcd670df0c8efd75d039cc951d80132"
+                }
+            ],
+            "buildsystem": "cmake",
```
Comment (hfiguiere) [inline com.solvespace.SolveSpace.json:129]: cmake-ninja doesn't work?

File: com.solvespace.SolveSpace.json [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+                {
+                    "type": "archive",
+                    "url": "http://ftp.gnome.org/pub/GNOME/sources/atkmm/2.28/atkmm-2.28.0.tar.xz",
+                    "sha256": "4c4cfc917fd42d3879ce997b463428d6982affa0fb660cafcc0bc2d9afcedd3a"
+                }
+            ]
+        },
+        {
+            "name": "gtkmm",
+            "config-opts": [],
```
Comment (hfiguiere) [inline com.solvespace.SolveSpace.json:109]:
```suggestion
```

File: com.solvespace.SolveSpace.json [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+                {
+                    "type": "archive",
+                    "url": "https://download.gnome.org/sources/libsigc++/2.10/libsigc%2B%2B-2.10.6.tar.xz",
+                    "sha256": "dda176dc4681bda9d5a2ac1bc55273bdd381662b7a6d49e918267d13e8774e1b"
+                }
+            ]
+        },
+        {
+            "name": "glibmm",
+            "config-opts": [],
```
Comment (hfiguiere) [inline com.solvespace.SolveSpace.json:58]:
```suggestion
```

**Comment by hfiguiere** [comment]
bot, build com.solvespace.SolveSpace

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
You might need to add `disable-shallow-clone` to the solvespace module for the checkout to work.

File: com.solvespace.SolveSpace.json (hunk 1 above)
Comment (rpavlik) [reply com.solvespace.SolveSpace.json:2]: ah, oops, didn't mean to commit that. It's a JSON manifest to get in-editor help.

File: com.solvespace.SolveSpace.json (hunk 2 above)
Comment (barthalion) [inline com.solvespace.SolveSpace.json:76]:
```suggestion
                    "url": "https://ftp.gnome.org/pub/GNOME/sources/cairomm/1.12/cairomm-1.12.0.tar.xz",
```

File: com.solvespace.SolveSpace.json (hunk 3 above)
Comment (barthalion) [inline com.solvespace.SolveSpace.json:89]:
```suggestion
                    "url": "https://ftp.gnome.org/pub/GNOME/sources/pangomm/2.40/pangomm-2.40.2.tar.xz",
```

File: com.solvespace.SolveSpace.json (hunk 6 above)
Comment (barthalion) [inline com.solvespace.SolveSpace.json:129]:
```suggestion
            "buildsystem": "cmake-ninja",
```

File: com.solvespace.SolveSpace.json (hunk 5 above)
Comment (barthalion) [inline com.solvespace.SolveSpace.json:141]:
```suggestion
            "buildsystem": "cmake-ninja",
```

**Comment by barthalion** [comment]
bot, build com.solvespace.SolveSpace

**Comment by rpavlik** [comment]
I'll take a look at this later this week.

**Comment by barthalion** [comment]
It needs some appdata fixes, other than this, it's good to be merged.

```
builddir/files/share/appdata/com.solvespace.SolveSpace.appdata.xml: FAILED:
• style-invalid         : Not enough <screenshot> tags, minimum is 1
```

**Comment by tinywrkb** [comment]
@rpavlik I've sent you a PR with an update to the submission.

**Comment by hfiguiere** [comment]
@tinywrkb you might as well resubmit it if you want to push for it.

**Comment by tinywrkb** [comment]
@hfiguiere there's a clearly an interest from a developer of the application, so it's much preferable to let the dev handle submission.

File: com.solvespace.SolveSpace.json (hunk 6 above)
Comment (rpavlik) [reply com.solvespace.SolveSpace.json]: fixed by @tinywrkb

File: com.solvespace.SolveSpace.json (hunk 5 above)
Comment (rpavlik) [reply com.solvespace.SolveSpace.json]: fixed by @tinywrkb

File: com.solvespace.SolveSpace.json (hunk 4 above)
Comment (rpavlik) [reply com.solvespace.SolveSpace.json]: Fixed

File: com.solvespace.SolveSpace.json (hunk 3 above)
Comment (rpavlik) [reply com.solvespace.SolveSpace.json]: fixed.

File: com.solvespace.SolveSpace.json (hunk 2 above)
Comment (rpavlik) [reply com.solvespace.SolveSpace.json]: fixed by @tinywrkb

**Comment by tinywrkb** [comment]
bot, build com.solvespace.SolveSpace

**Comment by rpavlik** [comment]
Yes, there was a willingness, just with a lack of time. Thanks very much to @tinywrkb for cleaning the manifest up a lot and updating it.

**Comment by rpavlik** [comment]
Hmm, absence of a screenshot. I can fix that going forward, but I'm not sure there's one in the repo/archive to point to for the existing release.

**Comment by tinywrkb** [comment]
Yes, only screenshot is missing according to `appstream-util validate`.

**Comment by rpavlik** [comment]
OK, so I have added screenshots upstream:

- waiting for  https://github.com/solvespace/solvespace-web/pull/36  to merge to actually be on the web
- waiting for https://github.com/solvespace/solvespace/pull/1026 to merge to have the updated metadata in the source

Not sure the best way to get this into the flathub manifest, since it's using an existing archive of the source. Is there an easy way to patch the metadata that gets installed? (There have been non-negligible commits to the git repo since 3.1 so just taking the master branch isn't a great solution.)

**Comment by tinywrkb** [comment]
Oh! Sorry for the wrong indent.

**Comment by tinywrkb** [comment]
bot, build com.solvespace.SolveSpace

File: com.solvespace.SolveSpace.json [hunk 9]
```diff
@@ -253,10 +253,14 @@
                         "url-query": ".assets[] | select(.name==\"solvespace-\" + $version + \".tar.xz\") | .browser_download_url"
                     }
                 },
+                {
+                    "type": "patch",
+                    "path": "metainfo-update-add-screenshots.patch"
+                }
```
Comment (tinywrkb) [inline com.solvespace.SolveSpace.json:259]: missing comma

**Comment by tinywrkb** [comment]
bot, build com.solvespace.SolveSpace

**Comment by rpavlik** [comment]
OK, cleaned up that patch.

**Comment by tinywrkb** [comment]
bot, build com.solvespace.SolveSpace

**Comment by tinywrkb** [comment]
@rpavlik the bot will respect the `bot, build ...` instruction if you post it here, so it fails, and you pushed a fix, then try again.  
I probably will only be able to look at this again much later today or tomorrow.  
If it builds successfully, then you can try asking for this to be reviewed by someone who can merge this at the [#flatpak Matrix room](https://matrix.to/#/#flatpak:matrix.org).

**Comment by rpavlik** [comment]
ah, very cool, didn't know if I had permissions to trigger the bot. Thanks, hopefully I can finish getting this ready to work myself.

**Comment by tinywrkb** [comment]
Looks good. I remember testing f-e-d-c properties, so the checker shows a warning possibly due to a temporary issue with `release-monitoring.org`.  
You can replace the check of eigen's source by switching to HTML checker, though I don't think this blocks approval.

```
diff --git a/com.solvespace.SolveSpace.json b/com.solvespace.SolveSpace.json
index d5f1f74..359dc78 100644
--- a/com.solvespace.SolveSpace.json
+++ b/com.solvespace.SolveSpace.json
@@ -199,8 +199,9 @@
                     "url": "https://gitlab.com/libeigen/eigen/-/archive/3.4.0/eigen-3.4.0.tar.gz",
                     "sha256": "8586084f71f9bde545ee7fa6d00288b264a2b7ac3607b974e54d13e7162c1c72",
                     "x-checker-data": {
-                        "type": "anitya",
-                        "project-id": 13751,
+                        "type": "html",
+                        "url": "https://eigen.tuxfamily.org/index.php?title=Main_Page",
+                        "version-pattern": "latest stable release</b> is Eigen ([\\d\\.-]+[\\d])",
                         "stable-only": true,
                         "url-template": "https://gitlab.com/libeigen/eigen/-/archive/$version/eigen-$version.tar.gz"
                     }
```

**Comment by tinywrkb** [comment]
To whoever merge this, the generated test build works great.  
Maybe send collaborator invites also to @phkahler and @ppd.

File: flathub.json [hunk 10]
```diff
@@ -0,0 +1,3 @@
+{
+    "require-important-update": true
```
Comment (hfiguiere) [inline flathub.json:2]: first time I see this...

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.solvespace.SolveSpace.json [hunk 11]
```diff
@@ -0,0 +1,272 @@
...
+        "--socket=fallback-x11",
+        "--socket=wayland"
+    ],
+    "cleanup": [
+        "/include",
+        "/lib/cmake",
+        "/lib/pkgconfig",
+        "/share/aclocal",
+        "/share/pkgconfig",
+        "*.la"
```
[the remark this answers is not in the document]
Comment (tinywrkb) [reply com.solvespace.SolveSpace.json:19]: > there is no *.a lying around ?

No, no static libs, explicitly disabled where needed.

> thre is also the developer documentation of the gtk bindings (can be built without) that might need to be cleaned up

No docs, maintainer_mode wasn't set in meson options, so docs are disabled by default.

File: flathub.json (hunk 10 above)
Comment (tinywrkb) [reply flathub.json:2]: Yes, it's new and shiny, and very much fitting here.

**Pull request closed without merging**